### PR TITLE
Support writing timestamp(0) as datetime type in ClickHouse

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

**Comment by ebyhr** [comment]
CI
* #9768

File: plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java [hunk 1]
```diff
@@ -573,6 +591,25 @@ private static void verifySupportedDate(long value)
...
+            verifySupportedTimestamp(epochSecond);
+            statement.setObject(index, LocalDateTime.ofEpochSecond(epochSecond, 0, UTC));
+        };
+    }
+
+    private static void verifySupportedTimestamp(long epochSecond)
+    {
+        if (epochSecond < MIN_SUPPORTED_TIMESTAMP_EPOCH || epochSecond > MAX_SUPPORTED_TIMESTAMP_EPOCH) {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss");
+            throw new TrinoException(INVALID_ARGUMENTS, format("Timestamp must be between %s and %s: %s", MIN_SUPPORTED_TIMESTAMP.format(formatter), MAX_SUPPORTED_TIMESTAMP.format(formatter), LocalDateTime.ofEpochSecond(epochSecond, 0, UTC).format(formatter)));
```
Comment (hashhar) [inline plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java:609]: If we remove validation on our side does ClickHouse return a useful error?

Or let's make the error message convey that ClickHouse supports timestamps in the specified range. Current message looks like Trino limitation.

**Review by hashhar — COMMENTED** [review]
The PR title sounds misleading. It seems we supported writing timestamp(0) but didn't have any validation (in `fromTrinoTimestamp`) to ensure that values which have higher precision are not allowed.

Looks good to me otherwise.

**Comment by ebyhr** [comment]
> we supported writing timestamp(0)

We didn't support creating such type. See https://github.com/trinodb/trino/pull/10751/files#diff-7ce7bdaa32b38505167fecaa1621f0e71d942c962a7f39c0cce4ea5e21ef0a1aR540-R542

File: plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java:609]: ClickHouse stores incorrect results for invalid timestamps. https://github.com/trinodb/trino/blob/master/plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java#L448

**Review by hashhar — APPROVED** [review]

**Pull request merged**